docs/tutorials/multimodal/clip_embedding.md

```diff
@@ -0,0 +1,115 @@
+        'https://farm4.staticflickr.com/3795/9591251800_9c9727e178_z.jpg',
+        'http://farm8.staticflickr.com/7188/6848765123_252bfca33d_z.jpg',
+        'https://farm6.staticflickr.com/5251/5548123650_1a69ce1e34_z.jpg']
+
+image_paths = [download(url) for url in urls]
+print(image_paths)
+```
+
+Let's extract some image embedding from the CLIP vision encoder,
+
+```{.python .input}
+from autogluon.multimodal import MultiModalPredictor
+predictor = MultiModalPredictor(hyperparameters={"model.names": ["clip"]}, problem_type="zero_shot")
+
+# extract image embeddings.
+image_embeddings = predictor.extract_embedding({"image": image_paths})
+for k, v in image_embeddings.items():
+    print(k, v.shape)  # image (5, 768)
+```
+
+The output has a shape of (5, 768), because there are 5 images, each of the image embedding has a dimension of 768. 
+
+Similarly, you can also extract text embeddings for other tasks, 
+
+```{.python .input}
+# extract text embeddings.
+text_embeddings = predictor.extract_embedding({"text": ['There is a carriage in the image']})
+for k, v in text_embeddings.items():
+    print(k, v.shape)  # text (1, 768)
+```
+
+Then you can use the embeddings for a range of tasks such as image retrieval, text retrieval, image-text retrieval and matching/ranking. 
+
+
+## Image retrieval by text query
+
+Suppose we have a large image database (e.g., video footage), now we want to retrieve some images defined by a text query. How can we do this? 
+
+It is simple. First, extract all the image embeddings either online or offline as shown above. Then, compute the text embedding of the text query. Finally, compute the cosine similarity between the text embedding and all the image embeddings and return the top candidates. 
+
+Let's reuse the example we have above. We already have 5 image embeddings in **image_embeddings**, and 1 text embedding in **text_embeddings**, now we normalize the feature and then compute their similarities,
+
+```{.python .input}
+image_features = image_embeddings['image']
+text_features = text_embeddings['text']
+
+from numpy import linalg as LA
+import numpy as np
+image_features /= LA.norm(image_features, axis=-1, keepdims=True)
+text_features /= LA.norm(text_features, axis=-1, keepdims=True)
+
+similarity = 100.0 * np.matmul(image_features, text_features.T)
+from scipy.special import softmax
+print(softmax(similarity))
+```
+
+We can see that we successfully find the image with a carriage in it. 
+
+```{.python .input}
+from IPython.display import Image, display
+pil_img = Image(filename=image_paths[2])
+display(pil_img)
+```
+
+If we want to switch to another text query, we simply re-compute text embeddings and do this similarity comparison again,
+
+```{.python .input}
+text_embeddings = predictor.extract_embedding({"text": ['There is an airplane over a car.']})
+
+text_features = text_embeddings['text']
+text_features /= LA.norm(text_features, axis=-1, keepdims=True)
```

Comment:
No need to normalize the features, which are already normalized inside the predictor.

Reply:
Cool, thanks